pyincore_data/utils/datautil.py

```diff
@@ -51,3 +62,177 @@ def convert_dislocation_pd_to_csv(in_pd, save_columns, programname, savefile):
         # Save cen_blockgroup dataframe with save_column variables to csv named savefile
         print("CSV data file saved to: " + programname + "/" + savefile + ".csv")
         in_pd[save_columns].to_csv(programname + "/" + savefile + ".csv", index=False)
+
+    @staticmethod
+    def get_features_by_fips(state_county_fips):
+        """
+        Downloads a GeoJSON feature collection from the NSI endpoint using the provided county FIPS code
+        and returns it as a GeoDataFrame with additional columns for FIPS, state FIPS, and county FIPS.
+
+        Args:
+            state_county_fips (str): The combined state and county FIPS code (e.g., '15005').
+
+        Returns:
+            gpd.GeoDataFrame: A GeoDataFrame containing the features with additional columns.
+        """
+        print("Requesting data for " + str(state_county_fips) + " from NSI endpoint")
+        json_url = cfg.NSI_URL_FIPS + str(state_county_fips)
+        result = requests.get(json_url)
+        result.raise_for_status()
+        result_json = result.json()
+
+        collection = FeatureCollection(result_json['features'])
+
+        gdf = gpd.GeoDataFrame.from_features(collection['features'])
+        gdf = gdf.set_crs(epsg=4326)
+
+        gdf = DataUtil.add_columns_to_gdf(gdf, state_county_fips)
+
+        return gdf
+
+    @staticmethod
+    def download_nsi_data_state_file(state_fips):
+        """
+        Downloads a zipped GeoPackage file for the given state FIPS code from the NSI endpoint.
+
+        Args:
+            state_fips (str): The state FIPS code (e.g., '29' for Missouri).
+
+        Returns:
+            None
+        """
+        file_name = cfg.NSI_PREFIX + str(state_fips) + ".gpkg.zip"
+        file_url = "%s/%s" % (cfg.NSI_URL_STATE, file_name)
+        print("Downloading NSI data for the state: " + str(state_fips))
+        r = requests.get(file_url, stream=True)
```

Comment:
Just curious, no need for `result.raise_for_status()` here?

Reply:
added